### PR TITLE
helm: add support for kube-version and add cli args for both kube-version and api-versions

### LINKED ISSUE #5112: Introduce Support for kube-version Parameter for Consuming Helm Charts Relying on Capabilities
### Eschewed features

- [X] This issue is not requesting templating, unstuctured edits, build-time side-effects from args or env vars, or any other eschewed feature.

### What would you like to have added?

`--kube-version` flag

example:
``` yaml
- name: reloader
  version: '0.0.113'
  repo: https://stakater.github.io/stakater-charts
  releaseName: reloader
  namespace: reloader
  kubeVersion: '1.21'
  valuesInline:
    # https://github.com/stakater/Reloader/blob/master/deployments/kubernetes/chart/reloader/values.yaml
    fullnameOverride: reloader
    reloader:
      deployment:
        replicas: 1
        resources:
          limits:
            cpu: "100m"
            memory: "512Mi"
          requests:
            cpu: "10m"
            memory: "128Mi"
      containerSecurityContext:
        capabilities:
          drop:
            - ALL
        allowPrivilegeEscalation: false
        readOnlyRootFilesystem: true
      service:
        port: 9090
      serviceMonitor:
        enabled: true
        namespace: reloader
````

### Why is this needed?

To be able to consume helm templates that rely on Capabilities the `--kube-version` flag has to be supported.

### Can you accomplish the motivating task without this feature, and if so, how?

N/A

### What other solutions have you considered?

N/A

### Anything else we should know?

_No response_

### Feature ownership

- [ ] I am interested in contributing this feature myself! 🎉

### PR DESCRIPTION
**Pull request by MrFreezeex**
Add basic --kube-version flag support and test and add Kustomize CLI args for both api-versions and kube-version

CLI args are useful since you could use one single kustomization file/helm chart for multiple clusters (which have different versions). Also it's easier to have those on the CLI if the user has some kind of tooling that will end up calling kustomize and that could pass those to Helm in the end (i.e.: ArgoCD is doing that for Helm so it could do that for Kustomize as well that will end up calling Helm).

Fixes #5112

### CODE REVIEW

**Comment by MrFreezeex** [comment]
Hi @koba1t, sorry for the ping. Since I saw you reviewing the helm OCI PR and as this is also a helm related PR, do you think it would it be possible for you to take a look/review this PR as well whenever you can? :pray:

**Comment by imranismail** [comment]
Hi @cailynse, sorry for the ping. But can you have a look at this, it's a key piece that's blocking the use of modern helm charts.

**Comment by koba1t** [comment]
Hi @MrFreezeex 
Thanks for your contribution!

Looks almost good, but I think more test cases.
Could you add any test for testing when output changes due to the `kubeVersion` being different?

**Comment by koba1t** [comment]
@MrFreezeex 
> Do you mean like a test when you pass both the CLI option to kustomize and kubeVersion to kustomization.yaml 

Hmm, I think we should add a test that covers such cases.

But I want to add any tests to check when the `KubeVersion` parameter changes. I think looks like it only tests for `v1.27`.
https://github.com/kubernetes-sigs/kustomize/pull/5270/files#diff-29c329264899fb54b29d61bc6f9a5db64d559c4c428c5ffafbc2bb76e43a1307R20
I think it better to check the output after exec `helm template` than to test the passing parameter. If you can, Could you change that?

**Comment by MrFreezeex** [comment]
> > Do you mean like a test when you pass both the CLI option to kustomize and kubeVersion to kustomization.yaml
> 
> Hmm, I think we should add a test that covers such cases.

Sure I can take a look :+1: 

> But I want to add any tests to check when the `KubeVersion` parameter changes. I think looks like it only tests for `v1.27`. https://github.com/kubernetes-sigs/kustomize/pull/5270/files#diff-29c329264899fb54b29d61bc6f9a5db64d559c4c428c5ffafbc2bb76e43a1307R20 I think it better to check the output after exec `helm template` than to test the passing parameter. If you can, Could you change that?

Hmm well I can do that but such test would be testing directly helm rather than a kustomize feature. In the tests of kustomize I currently test that if you pass `kubeVersion` it correctly set the appropriate helm flag + that if it's not set it doesn't pass the flag.

**Comment by MrFreezeex** [comment]
Actually there is one for apiversions as well even though it doesn't test much, will check how to do that :+1:

**Comment by MrFreezeex** [comment]
I added a new test with two different kubeVersion that make some change to the ingress group (`networking.k8s.io/v1beta1` and `networking.k8s.io/v1`) I believe that's what you were looking for?

File: api/krusty/helmchartinflationgenerator_test.go [hunk 1]
```diff
@@ -684,6 +684,200 @@ spec:
...
+	defer th.Reset()
+	if err := th.ErrIfNoHelm(); err != nil {
+		t.Skip("skipping: " + err.Error())
+	}
+
+	copyValuesFilesTestChartsIntoHarness(t, th)
+
+	th.WriteK(th.GetRoot(), `
+namespace: default
+helmCharts:
```
Comment (koba1t) [inline api/krusty/helmchartinflationgenerator_test.go:698]: Looks like two helms generating one time.
So, I think I prefer generating one `helmCharts` two times in this case due to eliminate side effects.

**Review by koba1t — CHANGES_REQUESTED** [review]
Thanks @MrFreezeex 
I added one minor suggestion for a test case.
Please check this.

File: api/krusty/helmchartinflationgenerator_test.go (hunk 1 above)
Comment (MrFreezeex) [reply api/krusty/helmchartinflationgenerator_test.go:698]: Sure, I think I did what you wanted! Let me know :+1:

**Review by koba1t — COMMENTED** [review]
@MrFreezeex 
Thanks for your contribution!

/approve
/lgtm

**Pull request merged**